parsers/FR_O.py

```diff
@@ -1,351 +1,188 @@
+        raise ParserException(
+            "FR_O.py",
+            f"Historical data not implemented for {zone_key} in this parser.",
+            zone_key,
+        )
+    elif target_datetime is None and zone_key not in LIVE_DATASETS.keys():
+        raise ParserException(
+            "FR_O.py",
+            f"Live data not implemented for {zone_key} in this parser.",
+            zone_key,
+        )
+
+    URL_QUERIES: Dict[str, Union[str, None]] = {
+        #   "refine": "statut:Validé" if target_datetime else None,
+        "timezone": "UTC",
+        "order_by": f"{DATE_STRING_MAPPING[zone_key]} desc",
+        "refine": f"{DATE_STRING_MAPPING[zone_key]}:{target_datetime.strftime('%Y')}"
+        if target_datetime
+        else None,
+    }
+
+    url = generate_url(zone_key, target_datetime)
+    response: Response = ses.get(url, params=URL_QUERIES)
+    data = response.json()
+    return data, DATE_STRING_MAPPING[zone_key]
 
 
-def fetch_price(
-    zone_key: str,
+def fetch_production(
+    zone_key: str = "FR-COR",
     session: Optional[Session] = None,
     target_datetime: Optional[datetime] = None,
-    logger: Logger = getLogger(__name__),
+    logger=getLogger(__name__),
 ):
-    if target_datetime is None:
-        raise NotImplementedError("There is no real time data")
-
-    r = session or requests.session()
-    params = get_param(zone_key, target_datetime)
-    api = get_api(zone_key, target_datetime)
-    date_name = get_date_name(zone_key, target_datetime)
-
-    r = session or requests.session()
-
-    # Data retrievement
-    response = r.get(api, params=params)
-    data = json.loads(response.content)
-    datapoints = []
-
-    if len(data["records"]) > 0 and ("fields" in data["records"][0]):
-        for i in range(0, len(data["records"])):
-            fields = data["records"][i]["fields"]
-            if "cout_moyen_de_production_eu_mwh" in fields:
-                datetime_result = arrow.get(fields[date_name]).datetime
-                datapoints.append(
-                    {
-                        "zoneKey": zone_key,
-                        "currency": "EUR",
-                        "datetime": datetime_result,
-                        "source": get_source(zone_key, target_datetime),
-                        "price": float(fields["cout_moyen_de_production_eu_mwh"]),
-                    }
+    data, date_string = fetch_data(zone_key, session, target_datetime, logger)
+    if data == []:
+        raise ParserException(
+            "FR_O.py",
+            f"No data available for {zone_key} for {target_datetime.strftime('%Y')}"
+            if target_datetime
+            else f"No live data available for {zone_key}.",
+            zone_key,
+        )
+    return_list: List[Dict[str, Any]] = []
+    for object in data:
+        production: Dict[str, Union[float, int]] = {}
+        storage: Dict[str, Union[float, int]] = {}
+        for key in object:
```

Comment:
Same here, key could be production_type ?

Reply:
Ah but I see that since it's mixed with storage it might be hard to find a better word.

Reply:
There isn't actally any storage in the parser yet but perhaps mode_key would be better?